lib/sentry.coffee

```diff
@@ -100,12 +100,15 @@ module.exports = class Sentry extends events.EventEmitter
 
     # Takes a function and produces a function that calls the given function, sending any errors it
     # produces to Sentry.
+    globals: {}
     wrap:
       if @enabled
-        (fn) -> (args..., cb) ->
-          fn args..., (err, results...) ->
+        (fn) -> (args..., cb) =>
+          fn args..., (err, results...) =>
             if err?
-              log_to_sentry err, {args}, (sentry_err) ->
+              extra = this.globals
```

Comment:
> If we wrap multiple functions will this cause problems?

The globals are part of the wrapper. You can create multiple wrappers - one for each function. 

> Can we change the interface to be something like

The extra context is not always known at the time of the wrapping, so we need a way to send them after the wrapped function started executing.